arch/risc-v/src/common/crt0.c

```diff
@@ -88,6 +88,71 @@ static void sig_trampoline(void)
   );
 }
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif
+
+/* Linker defined symbols to .ctors and .dtors */
+
+  extern void (*_sctors)(void);
+  extern void (*_ectors)(void);
+  extern void (*_sdtors)(void);
+  extern void (*_edtors)(void);
+
+#if defined(__cplusplus)
+}
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif
+
+/****************************************************************************
+ * Name: exec_ctors
+ *
+ * Description:
+ *   Call static constructors
+ *
+ ****************************************************************************/
+
+  static void exec_ctors(void)
+  {
+    for (void (**ctor)(void) = &_sctors; ctor != &_ectors; ctor++)
+      {
+        (*ctor)();
+      }
+  }
```

Comment:
```suggestion
static void exec_ctors(void)
{
  for (void (**ctor)(void) = &_sctors; ctor != &_ectors; ctor++)
    {
      (*ctor)();
    }
}
```

Reply:
does nxstyle complains here as well?

Reply:
BTW, why do we need to avoid name mangling for local functions?

Reply:
Yes it complains. 

I guess we don't need to avoid the name mangling, the end result for C++ should still work as these are never called across C/C++ and since they are local functions called locally the linker must keep them.

Reply:
yes, so maybe we can remove
```
extern "C"
{
```
wrapper?

Reply:
I can try that next week, I think it should work.

Reply:
Done. Seems to work / the name mangling does not seem to break anything.